src/main/resources/jupyter/edit-mode.js

```diff
@@ -0,0 +1,337 @@
+        'Run in Playground Mode': {
+            click: () => openPlaygroundMode(),
+            'class': 'btn-primary'
+        },
+        'Make a Copy': {
+            click: () => saveAs()
+        }
+    }
+
+    const noRemoteFileButtons = {
+        'Continue working': {
+            click: () => {},
+            'class': 'btn-primary'
+        }
+    }
+
+    const modeBannerId = "notification_mode"
+    const lockConflictTitle = "File is in use"
+    const syncIssueTitle = "File versions out of sync"
+    const syncIssueBody = "Your version of this file does not match the version in the workspace. What would you like to do?"
+    const syncIssueFatalBody = "This file was either deleted or never was stored with us."
+
+    //TODO URL resolution
+    const leoUrl = ''
+        // const leoUrl = 'http://localhost:8081'
+    const welderUrl = leoUrl + `/proxy/${googleProject}/${clusterName}/welder`
+    const localizeUrl = welderUrl + '/localize'
+        // const checkMetaUrl = welderUrl + '/checkMeta'
+    const checkMetaUrl = welderUrl + '/objects/metadata'
+    const lockUrl = welderUrl + '/lock'
+    const lastLockedTimer = 60000 // in ms, should be 60000 in final PR
+
+    const headers = {
+        'Content-Type': 'application/json',
+        'Access-Control-Allow-Headers': '*'
+    }
+
+    const basePayload = {
+        mode: "no-cors",
+        headers: headers
+    }
+
+    //TODO resolve these links
+    const jupyterBaseUrl = `/notebooks/${googleProject}/${clusterName}`
+    const jupyterContentsAPIUrl = jupyterBaseUrl + "/api/contents/"
+
+    const getLocalPath = () => {
+        return {
+            localObjectPath: Jupyter.notebook.notebook_path
+        }
+    }
+
+    function init() {
+        checkMeta()
+        initSyncMaintainer()
+    }
+
+    async function initSyncMaintainer() {
+        let syncMaintainer = setInterval(() => {
+            checkMeta()
+        }, lastLockedTimer)
+
+        window.onbeforeunload(function() {
+            clearInterval(syncMaintainer)
+        })
+    }
+
+    async function checkMeta() {
+        const payload = {
+            ...basePayload,
+            body: JSON.stringify(getLocalPath()),
+            method: 'POST'
+        }
+
+        return fetch(checkMetaUrl, payload)
+            .then(res => {
+                console.log('in first then of checkmeta', res)
+                if (!res.ok && res.status == 412) {
+                    promptUserWithModal(syncIssueTitle, noRemoteFileButtons, syncIssueFatalBody)
+                } else if (!res.ok && res.status >= 400) {
+                    throw Error(res.statusText)
+                }
+                return res.json()
+            })
+            .then(res => {
+                console.log('res in check meta after .json()', res)
+                handleCheckMetaResp(res) //sets meta state
+                toggleMetaFailureBanner(false) //sets banner for meta status
+                maintainLock(res) //gets lock 
```

Comment:
It does only get the lock if res.syncMode == 'Edit', but it needs to display a mode banner regardless, as we are collapsing that portion of the safe-mode extension into this extension to avoid the possibility of conflicting banners. I renamed the method to maintainLockState to clarify.

Reply:
OK - in my mind safe mode and locking are unrelated, so it doesn't make sense to me that `maintainLock` results in rendering of the banner (for edit or safe mode). I might suggest restructuring the tree of function calls here, but agreed it looks like the current end behavior is correct.